### DISCUSSION TITLE
cells with 100% contamination?

### DISCUSSION THREAD

**Discussion by plijnzaad** [campbio/celda#365] [Q&A]
Is it common for `decontx` to find many cells with nearly 100% contamination?  I get the following results (converted the contamination to percentage for readability).

```
quantile(result$decontX_contamination*100, probs=seq(0,1,0.1))
      =>
0%    0.00
10%   0.17
20%   0.41
30%   0.90
40%   2.02
50%   4.61
60%  10.43
70%  21.30
80%  34.85
90%  59.42
100% 99.95
```

This is using a vanilla call with no background and no labels, batch is simply the library. For one of the libraries there is a cluster with ~ 75% of the cells having > 90% contamination. Incidentally, there is a lot of erythrocyte contamination, but even if I discard all erythrocyte genes (not just hemoglobin genes), I still get a similar number of cells with very high contamination.

**Comment by joshua-d-campbell** [comment]
Hi @plijnzaad, thanks for the question. Do all of your cells in the dataset have close to 100% contamination or is it just the cells within specific clusters?

It may be helpful for us to see some of the diagnostic plots like `plotDecontXContamination`, `plotDimReduceCluster`, and `plotDecontXMarkerPercentage`. Sometimes the clustering can be an issue with very high levels of contamination and using an alternative set of cluster labels can help.

**Comment by plijnzaad** [comment]
Hi Joshua, 

thanks for your reply, I have attached a few images of 4 different libraries of related material, which was heavily contaminated  with erythroid cells and which I have tried to get rid in different ways before passing it to `decontx`. I have blanked out names of the markers/cell types in the `plotDecontXMarkerPercentage` plot (of yet another related library) as this is a bit sensitive. All  samples have a few fairly clear immune cell clusters, and a big sprawling cluster of different celltypes of our tissue of interest. 

clusters (decontx clustering) :

![Picture1](https://user-images.githubusercontent.com/11426231/159678236-e99f3d23-3ed9-447f-960d-1ff09c716dc1.png)
![Picture2](https://user-images.githubusercontent.com/11426231/159678234-41111057-5508-4c7d-abec-65492fda2415.png)
![Picture3](https://user-images.githubusercontent.com/11426231/159678232-759c3c10-bdfb-49fd-9fa7-237b38b68f94.png)

marker percentages:

![Picture4](https://user-images.githubusercontent.com/11426231/159678230-e0deae75-30df-4dc4-90c1-dfd3a1664d88.png)

before/after correction:

![Picture5](https://user-images.githubusercontent.com/11426231/159678220-55e221be-8f2b-464c-989f-6496c623f0e1.png)

**Comment by joshua-d-campbell** [comment]
Thanks for sending these along. Overall, I think the results seem in line with what we have seen previously. Many of the markers are reduced in certain clusters as observed in the percentage barplots (although it is a bit hard to tell since I am not sure which markers are supposed to be expressed or not in each cluster). 

There might be cases where better clustering would help. For example, in this sample where you see a group in which half of the cells have high contamination:

<img width="446" alt="image" src="https://user-images.githubusercontent.com/14353256/159916236-5db7ca0c-15b4-4b5b-9574-6b042ea9c4e8.png">

My guess is that these cells are all in the same cluster based on the density based clustering method DecontX relies on (but you can double check by plotting the `decontX_clusters` stored in the `colData` of the SCE object as described in the tutorial). If these cells were split into 2 clusters, then these cells may not have as high contamination estimates. You can try to tweak this in decontX by lowering the number of variable genes in the `varGenes` parameter (e.g. from 5000 by default to 2000) or by decreasing the clustering density parameter `dbscanEps` (e.g. from 1 to 0.5). You can also supply your own cluster labels via the `z` parameter. 

The other possibility is that these sets of cells are doublets. We have seen in the past that doublet clusters get flagged with relatively higher levels of contamination.
